x-pack/plugins/observability/public/hooks/use_fetch_triggered_alert_history.ts

```diff
@@ -0,0 +1,191 @@
+
+      const { totalTriggeredAlerts, histogramTriggeredAlerts, error, avgTimeToRecoverMS } =
+        await fetchTriggeredAlertsHistory({
+          http,
+          index,
+          ruleId,
+          signal: abortCtrlRef.current.signal,
+        });
+
+      if (error) throw error;
+      if (!isCancelledRef.current) {
+        setTriggeredAlertsHistory((oldState: TriggeredAlertsHistory) => ({
+          ...oldState,
+          triggeredAlertsData: {
+            totalTriggeredAlerts,
+            histogramTriggeredAlerts,
+            avgTimeToRecoverMS,
+          },
+          isLoadingRuleAlertsAggs: false,
+        }));
+      }
+    } catch (error) {
+      if (!isCancelledRef.current) {
+        if (error.name !== 'AbortError') {
+          setTriggeredAlertsHistory((oldState: TriggeredAlertsHistory) => ({
+            ...oldState,
+            isLoadingRuleAlertsAggs: false,
+            errorTriggeredAlertHistory: error,
+            triggeredAlertsData: undefined,
+          }));
+        }
+      }
+    }
+  }, [features, http, ruleId]);
+  useEffect(() => {
+    loadRuleAlertsAgg();
+  }, [loadRuleAlertsAgg]);
+
+  return triggeredAlertsHistory;
+}
+
+interface IndexName {
+  index: string;
+}
+
+export async function fetchIndexNameAPI({
+  http,
+  features,
+}: {
+  http: HttpSetup;
+  features: string;
+}): Promise<IndexName> {
+  const res = await http.get<{ index_name: string[] }>(`${BASE_RAC_ALERTS_API_PATH}/index`, {
+    query: { features },
+  });
+  return {
+    index: res.index_name[0],
+  };
+}
+
+export async function fetchTriggeredAlertsHistory({
+  http,
+  index,
+  ruleId,
+  signal,
+}: {
+  http: HttpSetup;
+  index: string;
+  ruleId: string;
+  signal: AbortSignal;
+}): Promise<FetchTriggeredAlertsHistory> {
+  try {
+    const res = await http.post<AsApiContract<any>>(`${BASE_RAC_ALERTS_API_PATH}/find`, {
+      signal,
+      body: JSON.stringify({
+        index,
+        size: 0,
+        query: {
+          bool: {
+            must: [
+              {
+                term: {
+                  [ALERT_RULE_UUID]: ruleId,
+                },
+              },
+              {
+                range: {
+                  [ALERT_TIME_RANGE]: {
+                    gte: 'now-30d',
+                    lt: 'now',
+                  },
+                },
+              },
+            ],
+          },
+        },
+        aggs: {
+          histogramTriggeredAlerts: {
+            date_histogram: {
+              field: ALERT_START,
+              fixed_interval: '1d',
+              extended_bounds: {
+                min: 'now-30d',
+                max: 'now',
+              },
+            },
+          },
+          avgTimeToRecoverMS: {
+            avg: {
+              script: {
+                source:
+                  "if (!doc['kibana.alert.end'].empty){return(doc['kibana.alert.end'].value.millis) - (doc['kibana.alert.start'].value.millis)}",
+              },
+            },
+          },
```

Comment:
There’s a field called `kibana.alert.duration.us` (`ALERT_DURATION`) which is essentially what the script is doing but in “microseconds” (.us suffix is micro seconds). Also I would filter by only `kibana.alert.status: recovered` so that you’re only displaying the average time for recovered alerts. You can either filter in the `query` section OR add a filter aggregation to limit the average to only `recovered` alerts. I would use the filter aggregation ONLY if you plan to include other metric aggregations in the date histogram, otherwise, just add it to the query section with the `ALERT_RULE_UUID`;

#### Example if you add the filter for "recovered" alerts to the query section
```
 avgTimeToRecoverMS: {
  avg: {
    field: ALERT_DURATION
  },
},
```
#### Example if you want to add the filter for "recovered" alerts
```
{
  avgTimeToRecoverMSE: {
    filter: {
      term: {
        [ALERT_STATUS]: 'recovered'
      }
    },
    aggs: {
      recoveryTime: {
        avg: {
          field: ALERT_DURATION
        }
      }
    }
  }
}
```
Keep in mind, scripted aggregations are EXTREMELY costly because it’s essentially doing the equivalent of a full table scan to create the value. We would want to avoid using scripted aggregations at all costs.